### PR TITLE
Improve the `get_processes` API and remove `pids()` method from `post/linux/priv`

### PR DESCRIPTION
**Pull request by pingport80**
## Summary
This PR improves the `get_processes` API for linux shell sessions by adding a way of getting the info from status files under `/proc` directory in case `ps_aux` fails.

This also removes a method `pids()` inside `post/linux/priv.rb` which was used to get the pids on the target system because `get_processes` can be used for the same task.

## Verification Steps
```
>> get_processes
=> 
[{"pid"=>1, "name"=>"systemd"},
 {"pid"=>9226, "name"=>"ssh-agent"},
..
..
..
 {"pid"=>1051, "name"=>"mate-settings-d"},
 {"pid"=>12331, "name"=>"bash"},
 {"pid"=>9908, "name"=>"sudo"},
 {"pid"=>9909, "name"=>"bash"}]
```

### CODE REVIEW

File: lib/msf/core/post/common.rb [hunk 1]
```diff
@@ -232,14 +232,29 @@ def get_processes
...
+        ps_aux.delete_at(0)
+        ps_aux.each do |p|
+          properties = p.split
+          process = {}
+          process['name'] = properties[10].gsub(/\[|\]/,"")
+          process['pid'] = properties[1].to_i
+          processes.push(process)
+        end
+      else
+        dir_proc = "/proc/"
```
Comment (bwatters-r7) [inline lib/msf/core/post/common.rb:246]: Let's check that this exists; guard against `nil` in each statement below.

Comment (pingport80) [reply lib/msf/core/post/common.rb:246]: okay, will do that. thanks!

Comment (smcintyre-r7) [reply lib/msf/core/post/common.rb:246]: It's a good idea to check that `/proc` exists but it should also handle cases where PIDs "vanish" during the course of iteration. This is listing the pids once and then iterating through them all, and it's possible that in a production environment the list may become stale if the connection is slow. A process list is a snapshot in time so I don't think it's a big deal if some entries are missing once all the information has been gathered, but if one process exits while this is iterating, we need to make sure that the whole thing won't fail.

Comment (pingport80) [reply lib/msf/core/post/common.rb:246]: Okay, I'll add some checks to avoid that 👍🏻

Comment (pingport80) [reply lib/msf/core/post/common.rb:246]: I've added a check if `/proc` exists and added checks in case some PIDs might vanish.

**Comment by smcintyre-r7** [comment]
This looks good to me, I pushed up some small changes in 525bdb4 that were mostly for style and extra rubocop apeasement. As for testing, I checked this on Fedora and Ubuntu without the `ps` binary to ensure it worked by taking the new code path. It's a bit slow but I don't think there's anyway around that now and since it's still favoring the output from `ps`, I think it's fine as is.

I'll land this momentarily, it looks like there's just a conflict that I need to resolve.

```
[{"pid"=>1, "name"=>"systemd"},
 {"pid"=>10, "name"=>"rcu_tasks_kthre"},
 {"pid"=>100, "name"=>"oom_reaper"},
 {"pid"=>1003, "name"=>"auditd"},
 {"pid"=>1005, "name"=>"sedispatch"},
 {"pid"=>101, "name"=>"writeback"},
 {"pid"=>102, "name"=>"kcompactd0"},
 {"pid"=>1029, "name"=>"rpciod"},
 {"pid"=>103, "name"=>"ksmd"},
 {"pid"=>1030, "name"=>"xprtiod"},
 {"pid"=>1036, "name"=>"ModemManager"},
 {"pid"=>1037, "name"=>"avahi-daemon"},
 {"pid"=>1038, "name"=>"earlyoom"},
 {"pid"=>104, "name"=>"khugepaged"},
 {"pid"=>1042, "name"=>"mcelog"},
 {"pid"=>1049, "name"=>"rngd"},
...
```

**Comment by smcintyre-r7** [comment]
# Release Notes
This improves the `get_processes` API on non-Windows systems with support that fails back to enumerating the `/proc` directory when the `ps` utility is not present.

**Pull request merged**